### ISSUE TITLE
Generator Node Not Updating

### ISSUE THREAD

**Issue by liamb315** [sharadmv/deepx#24]
Weights within components of a Generator node are not updating via training with RMSProp.  Specific example I'm using is,

Code:
generator = Generate(Vector(100) >> Repeat(LSTM(1024), 2) >> Softmax(100), args.sequence_length)
... 
gan = generator >> discriminator.right

gan.right.frozen = True
assert gan.get_parameters() == gan.left.get_parameters()
rmsprop_G = RMSProp(gan, ConvexSequentialLoss(CrossEntropy(), 0.5))_
